sasl.go

```diff
@@ -173,6 +177,112 @@ func (m *CramMD5Mechanism) step(challenge []byte) ([]byte, error) {
 	return append([]byte(fmt.Sprintf("%s ", m.username)), hash.Sum(nil)...), nil
 }
 
+// DigestMD5Mechanism corresponds to Digest-MD5 mechanism
+type DigestMD5Mechanism struct {
+	mechanismConfig *MechanismConfig
+	identity        string
+	username        string
+	password        string
+	host            string
+	nonceCount      int
+}
+
+// NewDigestMD5Mechanism returns a new DigestMD5Mechanism
+func NewDigestMD5Mechanism(username string, password string) *DigestMD5Mechanism {
+	return &DigestMD5Mechanism{
+		mechanismConfig: newDefaultConfig("DIGEST-MD5"),
+		username:        username,
+		password:        password,
+	}
+}
+
+func (m *DigestMD5Mechanism) start() ([]byte, error) {
+	return m.step(nil)
+}
+
+func (m *DigestMD5Mechanism) step(challenge []byte) ([]byte, error) {
+	if challenge == nil {
+		return nil, nil
+	}
+
+	// Create map of challenge
+	entries := strings.Split(string(challenge), ",")
```

Comment:
I think this way of parsing fails if some of the values have commas, for example, for this challenge:
```
nonce="rmD6R8aMYVWH+/ih9HGBr3xNGAR6o2DUxpKlgDz6gUQ=",
realm="example.org",qop="auth,auth-int,auth-conf",cipher="rc4-40,rc
4-56,rc4,des,3des",maxbuf=65536,charset=utf-8,algorithm=md5-sess
```
where `qop` has several values. This is why pure sasl uses a [custom parsing function](https://github.com/thobbs/pure-sasl/blob/master/puresasl/mechanisms.py#L339-L394).

Reply:
Ah this didn't occur to me, I am going to try it out, thanks for letting me know.

Reply:
And yes, I will gladly write a test :)